src/component/mxgraph/style/StyleManager.ts

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 Bonitasoft S.A.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import type { mxCell, mxGraphModel } from 'mxgraph';
+import { BpmnStyleIdentifier } from '.';
+import { setStyle } from './utils';
+import type { CssRegistry } from '../../registry/css-registry';
+
+export class StyleManager {
+  /**
+   * Contains the style of each cell.
+   */
+  private stylesByCell: Map<mxCell, string> = new Map();
+
+  /**
+   * Creates a new instance of the StyleManager.
+   *
+   * @param cssRegistry The CSS registry to get class names from.
+   * @param model The mxGraphModel instance.
+   */
+  constructor(readonly cssRegistry: CssRegistry, readonly model: mxGraphModel) {}
+
+  /**
+   * Deletes all saved styles.
+   */
+  clear(): void {
+    this.stylesByCell.clear();
+  }
+
+  /**
+   * Resets the style of a cell and applies the CSS classes.
```

Comment:
**nitpick:** wrong JSDoc, we don't manage CSS classes here

Reply:
It's not a wrong JSDoc, we really rebuild the style by adding the stored CSS classes to it.

Reply:
My bad 😸 This is important to keep this information.